formats/developers/compression.txt

```diff
@@ -0,0 +1,170 @@
+|                           |                            | decoded from base64.                                                  |
++---------------------------+----------------------------+-----------------------------------------------------------------------+
+| OME-XML, zipped           | Great – 124,764 bytes      | Worst – Entire dataset must be uncompressed, then the plane must be   |
+|                           |                            | decoded from base64.                                                  |
++---------------------------+----------------------------+-----------------------------------------------------------------------+
+| OME-XML, 7-zipped         | Great – 125,830 bytes      | Worst – Entire dataset must be uncompressed, then the plane must be   |
+|                           |                            | decoded from base64.                                                  |
++---------------------------+----------------------------+-----------------------------------------------------------------------+
+
+The performance penalty for accessing individual image planes from
+externally compressed formats (zip, gzip, and 7-zip) is high, since the
+data must be decompressed. There is little penalty for accessing them
+from uncompressed OME-XML or OME-TIFF—with OME-XML, file readers can
+build a list of offsets by skipping past the bulk of the BinData
+characters according to the Length attribute values, and with OME-TIFF,
+file readers can seek to the offsets indicated in the IFD entries.
+
+Accessing them from an uncompressed OME-TIFF file, however, is
+efficient. In addition, the TIFF format maintains compatibility with the
+multitude of existing software that works with single- and multi-page
+TIFF files.
+
+Space required on disk
+----------------------
+
+As shown in the table above, and the chart :ref:`figure-tiff-chart`, our 
+figures indicate that the most efficient format for space is OME-TIFF 
+compressed with the `7-zip <http://www.7-zip.org/>`_ utility. Also good are 
+gzipped OME-XML and zipped OME-TIFF.
+
+.. _figure-tiff-chart:
+
+.. figure:: /images/ome-tiff-chart.png
+   :align: center
+   :alt: OME-TIFF space used
+
+   OME-TIFF space used
+
+Uncompressed OME-TIFF format, while the least space-efficient, provides
+its own advantages: it is highly compatible and provides efficient
+access to individual image planes.
+
+OME-TIFF with LZW often performs nearly as well as the externally
+compressed formats (zip, gzip and 7-zip), without the performance
+penalty of searching through a compressed archive. However, LZW is
+noticeably less efficient to decode than uncompressed TIFF is, and LZW
+is an additional requirement for client software—it may be that some
+software supports uncompressed multi-page TIFF, but not LZW. Even more
+unfortunate, the 7-zip algorithm appears to perform less well on
+LZW-compressed OME-TIFFs than on uncompressed OME-TIFFs.
+
```

Comment:
It may also be worth mentioning here that some filesystems e.g. NTFS, ZFS can support transparent lossless compression on disk https://pthree.org/2012/12/18/zfs-administration-part-xi-compression-and-deduplication/ .  In the case of ZFS, this can actually improve performance when limited by disk bandwidth.  This means you can eliminate the need for external and internal compression of the files, at the expense of less efficient compression but with the gain of complete transparency and potentially improved performance.